pow/src/main/java/tech/pegasys/teku/pow/Web3jEth1Provider.java

```diff
@@ -58,17 +58,20 @@ public class Web3jEth1Provider extends AbstractMonitorableEth1Provider {
   private final Web3j web3j;
   private final AsyncRunner asyncRunner;
   private final LabelledMetric<Counter> requestCounter;
+  private final boolean trustedWeb3j;
 
   public Web3jEth1Provider(
       final SpecConfig config,
       final MetricsSystem metricsSystem,
       final String id,
       final Web3j web3j,
+      final boolean trustedWeb3j,
       final AsyncRunner asyncRunner,
       final TimeProvider timeProvider) {
     super(timeProvider);
     this.config = config;
     this.web3j = web3j;
+    this.trustedWeb3j = trustedWeb3j;
```

Comment:
I'm not sure I'd call this a trusted web3j. "Trusted" suggests there's something security related.  I wonder if we can just try to execute the `eth_chainId` method and if it's not supported skip the check rather than failing?